### PR TITLE
Remove stuff scheduled for deprecation

### PR DESCRIPTION
**Pull request by data-man**
I didn't find any tests for `macros.basename`, so I doubt in it.

### CODE REVIEW

**Comment by data-man** [comment]
Depends on #9837.

File: lib/core/macros.nim [hunk 1]
```diff
@@ -590,16 +537,6 @@ proc newCall*(theProc: NimNode,
...
-proc newCall*(theProc: NimIdent,
-              args: varargs[NimNode]): NimNode {.compileTime, deprecated.} =
-  ## produces a new call node. `theProc` is the proc that is called with
-  ## the arguments ``args[0..]``.
-  ## **Deprecated since version 0.18.1**; Use ``newCall(string, ...)``,
-  ## or ``newCall(NimNode, ...)`` instead.
-  result = newNimNode(nnkCall)
-  result.add(newIdentNode(theProc))
-  result.add(args)
-
```
Comment (mratsim) [inline lib/core/macros.nim]: I don't see a `newCall(theProc: nimNode, ...)`?

File: lib/pure/math.nim [hunk 2]
```diff
@@ -370,12 +370,6 @@ when not defined(JS): # C
     proc gamma*(x: float32): float32 {.importc: "tgammaf", header: "<math.h>".}
     proc gamma*(x: float64): float64 {.importc: "tgamma", header: "<math.h>".}
       ## Computes the the `gamma function <https://en.wikipedia.org/wiki/Gamma_function>`_ for ``x``.
-    proc tgamma*(x: float32): float32
-      {.deprecated: "use gamma instead", importc: "tgammaf", header: "<math.h>".}
-    proc tgamma*(x: float64): float64
-      {.deprecated: "use gamma instead", importc: "tgamma", header: "<math.h>".}
-      ## The gamma function
-      ## **Deprecated since version 0.19.0**: Use ``gamma`` instead.
```
Comment (mratsim) [inline lib/pure/math.nim]: Too early to remove?

File: lib/pure/math.nim [hunk 3]
```diff
@@ -463,11 +457,6 @@ when not defined(JS): # C
       ##  echo trunc(PI) # 3.0
       ##  echo trunc(-1.85) # -1.0
 
-  proc fmod*(x, y: float32): float32 {.deprecated: "use mod instead", importc: "fmodf", header: "<math.h>".}
-  proc fmod*(x, y: float64): float64 {.deprecated: "use mod instead", importc: "fmod", header: "<math.h>".}
-    ## Computes the remainder of ``x`` divided by ``y``.
-    ## **Deprecated since version 0.19.0**: Use the ``mod`` operator instead.
```
Comment (mratsim) [inline lib/pure/math.nim]: Too early to remove?

File: lib/pure/osproc.nim [hunk 4]
```diff
@@ -335,19 +335,6 @@ proc execProcesses*(cmds: openArray[string],
...
-  ## Specify -1 for no timeout. Returns the number of processes that are
-  ## ready to read from. The processes that are ready to be read from are
-  ## removed from `readfds`.
-  ##
-  ## **Warning**: This function may give unexpected or completely wrong
-  ## results on Windows.
-  ##
-  ## **Deprecated since version 0.17.0**: This procedure isn't cross-platform
-  ## and so should not be used in newly written code.
-
```
Comment (timotheecour) [inline lib/pure/osproc.nim]: please dont deprecate without something that can exactly replace this!

**Comment by Araq** [comment]
Yeah sorry, it's too early for this.

**Comment by Araq** [comment]
Well removing

sequtils.map (Deprecated since version 0.12.0:)
sequtils.mapIt (Deprecated since version 0.12.0:)

is ok.

**Comment by dom96** [comment]
what would be really useful is if you could remove the old sockets modules and their dependencies

**Comment by data-man** [comment]
@dom96 
You mean
deprecated\pure\
asyncio.nim
ftpclient.nim
rawsockets.nim
sockets.nim
?
Or whole deprecated\pure directory ? (excluding ospaths.nim maybe)

**Pull request closed without merging**